### ISSUE TITLE
Fail download Data to wifi

### ISSUE THREAD

**Issue by RemyBarbosa** [tonyofrancis/Fetch#243]
Hi ! i got an error when i try to download a file with data's mobile and i switch on wifi.
Can i do something for that or it come from the library ?

**Comment by tonyofrancis** [comment]
@RemyBarbosa You have to enabled try on network fail in the FetchConfiguration Builder.
```java
 final FetchConfiguration fetchConfiguration = new FetchConfiguration.Builder(this)
                .enableRetryOnNetworkGain(true)
                .build()
```
Please have a look at the sample app.

**Comment by Karuhanga** [comment]
Are you sure you have not restricted the download to WiFi only?

Lincoln Karuhanga

-----Original Message-----

**Comment by RemyBarbosa** [comment]
Yep the download start with mobile data, i can see the progression and if i switch to wifi during the download the i got an error.

**Comment by tonyofrancis** [comment]
@RemyBarbosa when the switch occurs an error will be thrown. But fetch will wait and retry again in 10 seconds. Are you setting a NetworkType on the request or Fetch? eg:
```java
fetch.setGlobalNetworkType(NetworkType.WIFI_ONLY);
//or
request.setNetworkType(NetworkType.WIFI_ONLY);
```
